### ISSUE TITLE
multi-label train

### ISSUE THREAD

**Issue by m66505196** [ultralytics/ultralytics#18797]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

hello, how to train YOLOv11 with <Class_ID> <x_center> <y_center> <width> <height> <feature_1> <feature_2> ... <feature_n> format?
nc: <count features>
names: ['glasses', 'hat', 'old', 'young'] ?
Do we need to make changes to the code? YOLOv5 or ... support? my task is detection with muti-label

**Comment by UltralyticsAssistant** [comment]
👋 Hello @m66505196, thank you for your interest in Ultralytics 🚀! We recommend checking out our [Docs](https://docs.ultralytics.com/) for in-depth guidance, including examples for [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage, which may address your question.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it effectively.

If this is a custom training ❓ Question, such as yours, we ask for more details to better assist you:
- Any modifications you’ve attempted in the code
- The exact format of your dataset annotations
- A snippet of your configurations (e.g., `data.yaml`)
- Training logs or error messages (if you’ve encountered any)

For multi-label applications or custom feature-based tasks, exploring our [Model Training Tips](https://docs.ultralytics.com/guides/model-training-tips/) might also offer some useful insights.

We’d like to highlight that you’re running on the open-source project YOLOv11 😉, which is outside the usual single-object detection scope, so some code modifications may indeed be necessary for custom tasks. Please ensure you’ve reviewed and are familiar with the [codebase](https://github.com/ultralytics/ultralytics). 

## Upgrade

Upgrade to the latest `ultralytics` package along with all dependencies in a [**Python>=3.8**](https://www.python.org/) environment to ensure you’re leveraging the latest features and fixes. You can do so with:

```bash
pip install -U ultralytics
```

## Environments

YOLO can be executed across multiple environments. Feel free to choose one of the following to fit your needs:
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** VM: See our [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI: Learn more in our [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**: [Quickstart guide here](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) and official image <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Community Support

Got more questions? You’re invited to join and contribute to discussions in our Ultralytics community:
- Check out our [Discord server](https://discord.com/invite/ultralytics) 🎧 for real-time interaction
- Browse and contribute to discussions on [Discourse](https://community.ultralytics.com/)
- Head to our [Subreddit](https://reddit.com/r/Ultralytics) to share ideas with users globally

## Status

The Ultralytics CI tests monitor the health of the project. As of the last 24-hour cycle, the current status is:  
<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

An Ultralytics engineer will review and assist you further soon. Thanks for reaching out and happy training! 🚀

**Comment by glenn-jocher** [comment]
@m66505196 thank you for your question! YOLO11 supports multi-label training when using additional features or attributes in the dataset. However, the standard YOLO format (`<Class_ID> <x_center> <y_center> <width> <height>`) doesn't natively support extra features like `<feature_1> <feature_2> ...`. To include such features, you would need to modify the dataset parsing and the model's training pipeline in the code.

For multi-label classification tasks, consider using the **classification mode** instead. You can define multiple labels for each class in the dataset YAML file (`names`), and train accordingly. Refer to the [Ultralytics Classification Docs](https://docs.ultralytics.com/tasks/classify) for more details.

If your task requires detection with additional labels, you might need to customize the code. For support beyond YOLO11's standard features, you could adapt the codebase or explore other YOLO versions like YOLOv5, which supports modifications.

**Comment by FiksII** [comment]
> For multi-label classification tasks, consider using the **classification mode** instead. You can define multiple labels for each class in the dataset YAML file (`names`), and train accordingly. Refer to the [Ultralytics Classification Docs](https://docs.ultralytics.com/tasks/classify) for more details.

Excuse me, but I can't find anything related to the multi-label classification. Only video contains some "clue", that same images might be in a different folders (but, I think it's just a mistake in the presentation).

**Comment by glenn-jocher** [comment]
You're right, the documentation doesn't explicitly cover multi-label classification in detail. Let me clarify:

For multi-label classification in YOLO11, you would need to modify the dataset loading process to handle multiple labels per image. This requires customizing the `update_labels_info` method in `ultralytics/data/dataset.py` to process multiple class labels per instance.

A simpler approach is to use a one-hot encoding system where each attribute (glasses, hat, old, young) becomes its own binary classification task. You could implement this by:

1. Creating a custom dataset class extending `YOLODataset`
2. Modifying the label format to support multiple positive classes per image
3. Adjusting the loss function to use binary cross-entropy instead of standard classification loss

Would you like me to provide more specific code guidance on implementing this approach? The current implementation would require these customizations as multi-label classification isn't directly supported in the standard pipeline.
